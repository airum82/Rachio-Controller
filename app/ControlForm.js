import React, { Component } from 'react';

class ControlForm extends Component {
  constructor() {
    super()
    this.state = {
      sortOrder: [],
      zones: [],
      type: '',
      duration: '0',
      zoneId: ''
    }
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    this.setState({
      [e.target.name] : e.target.value
    })
  }

  render() {
    return (
      <div className="form">
        <form onSubmit={(event) => this.props.runAllZones(this.state.duration, event)}>
          <input type="text" name="duration" onChange={(e) => this.handleChange(e)}/>
          <button>run all zones</button>
        </form>
      </div>
    )
  }
}

export default ControlForm;