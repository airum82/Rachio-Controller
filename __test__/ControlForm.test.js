import React from 'react';
import ControlForm from '../app/ControlForm';
import { shallow } from 'enzyme';

describe('ControlForm', () => {
  const mockRunAllZones = jest.fn();
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ControlForm
      runAllZones={mockRunAllZones}
    />)
  })
  it('should match snapshot upon render', () => {
    expect(wrapper).toMatchSnapshot();
  })
  it('handleChange should call set state with correct params', () => {
    const spy = jest.spyOn(wrapper.instance(), 'setState');
    const mockEvent = {
      target: {
        name: 'sarah',
        value: 'so much'
      }
    };
    wrapper.instance().handleChange(mockEvent);
    expect(spy).toHaveBeenCalledWith({ sarah: 'so much' })
  })
  it('form should call runAllZones on submit', () => {
    const mockEvent = {
      preventDefault: jest.fn()
    }
    const mockDuration = wrapper.state().duration;
    wrapper.find('form').simulate('submit', mockEvent);
    expect(mockRunAllZones).toHaveBeenCalledWith(mockDuration, mockEvent);
  })
})