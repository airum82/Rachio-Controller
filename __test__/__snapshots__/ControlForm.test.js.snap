// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlForm should match snapshot upon render 1`] = `
<div
  className="form"
>
  <h2>
    duration:
  </h2>
  <form
    className="control-form"
    onSubmit={[Function]}
  >
    <input
      name="duration"
      onChange={[Function]}
      type="text"
    />
    <button>
      run all zones
    </button>
  </form>
</div>
`;
